### ISSUE TITLE
Contentlets lose inherited permissions until cache is flushed

### ISSUE THREAD

**Issue by craigWagner99** [dotCMS/core#19608]
As reported in https://dotcms.zendesk.com/agent/tickets/102941

dotCMS v 5.3.8

There is some condition that is causing content to lose all permissions making them inaccessible to anyone but admins.

This happens to file assets and contentlets.

There does not appear to be an error logged when it happens.

Flushing cache appears to bring the correct permissions back.

**To Reproduce**
Steps to reproduce the behavior are still being researched.

**Screenshots**
CONTENTLET WITH LOST PERMISSIONS
![image](https://user-images.githubusercontent.com/49727631/99702051-f58cca00-2a5a-11eb-85c4-317aa80a568a.png)

FILE ASSET WITH LOST PERMISSIONS
![image](https://user-images.githubusercontent.com/49727631/99702117-09383080-2a5b-11eb-8780-cb05b809c614.png)

CONTENTLET AFTER FLUSHING CACHE
![image](https://user-images.githubusercontent.com/49727631/99702298-37b60b80-2a5b-11eb-9f33-cc5949f1994c.png)

FILE ASSET AFTER FLUSHING CACHE
![image](https://user-images.githubusercontent.com/49727631/99702173-19501000-2a5b-11eb-955c-b5df20697050.png)

**Comment by john-thomas-dotcms** [comment]
Note: From the screenshots above, you can see that when this happens, the "Permission Individually" button doesn't even display any more.  The button comes back after flushing the cache.

As discussed in the PO meeting, we're working to get more information on what is happening when this problem occurs:
1. Check permission reference in the DB to make sure it's correct.
    * For the content with the problem, and for any folders in the tree.
2. Flush just the permission entry from the cache (to prevent it from being cleared from the DB table).
    * We'll need a JSP to be able to do this.

**Comment by wezell** [comment]
Seen in a customer environment 
**Before cache flush**
```
dotcms=> select * from permission_reference where asset_id='2e91b093-0fe9-4d50-b4ca-1f9f6b2c82cf';
 id | asset_id | reference_id | permission_type
----+----------+--------------+-----------------
(0 rows)
```

**After cache flush**
```
dotcms=> select * from permission_reference where asset_id='2e91b093-0fe9-4d50-b4ca-1f9f6b2c82cf';
   id    |               asset_id               | reference_id |                     permission_type
---------+--------------------------------------+--------------+---------------------------------------------------------
 1015532 | 2e91b093-0fe9-4d50-b4ca-1f9f6b2c82cf | SYSTEM_HOST  | com.dotmarketing.portlets.htmlpageasset.model.IHTMLPage
```

**Comment by x0rsw1tch** [comment]
Hello, just wanted to relay some information about an instance we had with a client just recently... Here's the (non-precise) steps that led up to a particular page asset losing inherited permissions:

1. Open site browser
1. Open page preview
1. Make no changes to the page
1. Publish the page for funsies (Maybe, we're not sure if the page was re-pub'd before the next step)
1. Go back to site browser
1. Push publish page
1. Page loses permissions

**Comment by wezell** [comment]
Commit here: 81c1e6721a

**Comment by nollymar** [comment]
PR for version 20.12: https://github.com/dotCMS/core/pull/19729

**Comment by nollymar** [comment]
Passed internal qa

**Comment by bryanboza** [comment]
Fixed, tested on release-21.01 // Postgres // FF. Unable to reproduce after the changes

**Comment by dsilvam** [comment]
Related PR: https://github.com/dotCMS/core/pull/19722
